### PR TITLE
chore: update employee swagger document

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] update employee swagger

### CODE REVIEW

File: pkg/handler/employee/employee.go [hunk 1]
```diff
@@ -330,12 +356,13 @@ func (h *handler) UpdateGeneralInfo(c *gin.Context) {
...
 // @Summary Create new employee
 // @Description Create new employee
+// @id createEmployee
 // @Tags Employee
 // @Accept  json
 // @Produce  json
-// @Param Body body request.CreateEmployeeInput true "Body"
 // @Security BearerAuth
-// @Success 200 {object} view.EmployeeData
+// @Param Body body CreateEmployeeRequest true "Body"
```
Comment (hieuphq) [inline pkg/handler/employee/employee.go:364]:
```suggestion
// @Param Body body CreateEmployeeQuery true "Body"
```

File: pkg/handler/employee/employee.go [hunk 2]
```diff
@@ -395,13 +422,14 @@ func (h *handler) Create(c *gin.Context) {
...
 // @Description Update Skill for employee by id
+// @id updateSkills
 // @Tags Employee
 // @Accept  json
 // @Produce  json
 // @Param id path string true "Employee ID"
-// @Param Body body request.UpdateSkillsInput true "Body"
 // @Security BearerAuth
-// @Success 200 {object} view.UpdateSkillsEmployeeResponse
+// @Param Body body UpdateSkillsRequest true "Body"
```
Comment (hieuphq) [inline pkg/handler/employee/employee.go:431]:
```suggestion
// @Param Body body UpdateSkillsQuery true "Body"
```

File: pkg/handler/employee/employee.go [hunk 3]
```diff
@@ -447,13 +475,14 @@ func (h *handler) UpdateSkills(c *gin.Context) {
...
 // @Description Update personal info of the employee by id
+// @id updatePersonalInfo
 // @Tags Employee
 // @Accept  json
 // @Produce  json
 // @Security BearerAuth
 // @Param id path string true "Employee ID"
-// @Param Body body request.UpdatePersonalInfoInput true "Body"
-// @Success 200 {object} view.UpdatePersonalEmployeeResponse
+// @Param Body body UpdatePersonalInfoRequest true "Body"
```
Comment (hieuphq) [inline pkg/handler/employee/employee.go:484]:
```suggestion
// @Param Body body UpdatePersonalInfoQuery true "Body"
```

**Review by hieuphq — CHANGES_REQUESTED** [review]

File: pkg/handler/employee/employee.go [hunk 4]
```diff
@@ -294,11 +298,33 @@ func (h *handler) UpdateGeneralInfo(c *gin.Context) {
...
+			return
+		}
+		OrganizationIDs = append(OrganizationIDs, organizationID)
+	}
+
+	referredBy, err := model.UUIDFromString(body.ReferredBy)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, errs.ErrInvalidReferredBy, nil, ""))
+		return
+	}
```
[the remark this answers is not in the document]
Comment (datphamcode295) [reply pkg/handler/employee/employee.go]: I already added view.UUID to replace model.UUID to solve this

File: pkg/view/pagination.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+var maxPageSize int64 = 999
+
+type SortOrder string
+
+type Pagination struct {
+	Page         int64  `json:"page" form:"page,default=0"`            // page index
+	Size         int64  `json:"size" form:"size"`                      // page size
+	Sort         string `json:"sort" form:"sort" swaggerignore:"true"` // sort by field
+	Standardized bool   `json:"-" form:"-" swaggerignore:"true"`
```
Comment (hieuphq) [inline pkg/view/pagination.go:17]: ??

File: pkg/view/pagination.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+		operator := "ASC"
+		if sort[0] == '-' {
+			operator = "DESC"
+			sort = strings.Replace(sort, "-", "", 1)
+		}
+		sorts[i] = fmt.Sprintf("%s %s", sort, operator)
+	}
+
+	return strings.Join(sorts, ",")
+}
```
Comment (hieuphq) [inline pkg/view/pagination.go:86]: ??

**Review by hieuphq — CHANGES_REQUESTED** [review]

File: pkg/view/pagination.go (hunk 6 above)
Comment (datphamcode295) [reply pkg/view/pagination.go:86]: này bên handler có dùng á anh

File: pkg/controller/employee/list.go [hunk 7]
```diff
@@ -2,6 +2,7 @@ package employee
 
 import (
 	errors "errors"
+
```
Comment (hieuphq) [inline pkg/controller/employee/list.go:5]: ??

File: pkg/controller/employee/list.go [hunk 8]
```diff
@@ -2,6 +2,7 @@ package employee
 
 import (
 	errors "errors"
```
Comment (hieuphq) [inline pkg/controller/employee/list.go:4]: ??

File: pkg/handler/employee/employee.go [hunk 9]
```diff
@@ -45,12 +45,13 @@ func New(controller *controller.Controller, store *store.Store, repo store.DBRep
...
 // @Summary Get the list of employees
 // @Description Get the list of employees with pagination and workingStatus
+// @id getEmployeeList
 // @Tags Employee
 // @Accept  json
 // @Produce  json
 // @Security BearerAuth
-// @Param Body body request.GetListEmployeeInput true "Body"
-// @Success 200 {object} view.EmployeeListDataResponse
+// @Param Body body GetListEmployeeInput true "Body"
```
Comment (hieuphq) [inline pkg/handler/employee/employee.go:53]:
```suggestion
// @Param Body body GetListEmployeeQuery true "Body"
```

File: pkg/handler/employee/employee.go [hunk 10]
```diff
@@ -592,12 +605,13 @@ func (h *handler) UploadAvatar(c *gin.Context) {
...
 // @Summary Update role by employee id
 // @Description Update role by employee id
+// @id updateRole
 // @Tags Employee
 // @Accept  json
 // @Produce  json
 // @Security BearerAuth
 // @Param id path string true "Employee ID"
-// @Param roleID body model.UUID true "Account role ID"
+// @Param roleID body string true "Account role ID"
```
Comment (hieuphq) [inline pkg/handler/employee/employee.go:614]:
```suggestion
// @Param roleID body UpdateRoleRequest true "Account role ID"
```

**Review by hieuphq — CHANGES_REQUESTED** [review]

**Review by hieuphq — APPROVED** [review]

**Pull request merged**